tiledb/ml/_parallel_utils.py

```diff
@@ -0,0 +1,36 @@
+# type: ignore
+from concurrent.futures import Future, ThreadPoolExecutor
+from typing import List
+
+import tiledb
+
+
+def batch_slice_func(
```

Comment:
- This is a private function, so prefix with `_`
- No need for `_func` suffix
- A 10 line private function for one line of code is rather low signal-to-noise ratio; not every tiny helper needs a docstring.